tripperware_test.go

```diff
@@ -4,11 +4,43 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"net/http/httptest"
+	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/gol4ng/httpware"
+	"github.com/gol4ng/httpware/mocks"
 )
 
+func TestRoundTripFunc_RoundTrip(t *testing.T) {
+	var roundTripperMock = &mocks.RoundTripper{}
+	var req *http.Request
+	var resp = &http.Response{
+		Status:           "OK",
+		StatusCode:       http.StatusOK,
+		ContentLength:    30,
+	}
+
+	req = httptest.NewRequest(http.MethodGet, "http://fake-addr", nil)
+	// mock roundTripper calls
+	roundTripperMock.On("RoundTrip", req).Return(resp, nil)
+	http.DefaultTransport = roundTripperMock
```

Comment:
this global var is well overridden but [this part of code is not called](https://github.com/gol4ng/httpware/blob/master/tripperware.go#L27) even if `tripper` is equal to `nil`